File: backend/src/main/java/br/com/transoft/backend/dto/route/RoutePresenter.java
package br.com.transoft.backend.dto.route;

import br.com.transoft.backend.dto.driver.DriverPresenter;
import br.com.transoft.backend.dto.school.SchoolPresenter;
import br.com.transoft.backend.dto.vehicle.presenter.VehiclePresenter;

public record RoutePresenter(
        String routeId,
        String name,
        Boolean active,
        SchoolPresenter school,
        DriverPresenter defaultDriver,
        VehiclePresenter defaultVehicle,
        DepartureTripDto departureTrip,
        ReturnTripDto returnTrip,
        DaysOfWeekDto daysOfWeek
) {}

File: frontend-web/src/models/route/RoutePresenter.ts
import type DriverPresenter from "../driver/DriverPresenter";
import type SchoolPresenter from "../SchoolPresenter";
import type VehiclePresenter from "../vehicle/VehiclePresenter";

export default interface RoutePresenter {
    routeId: string;
    name: string;
    active: boolean;
    school: SchoolPresenter;
    defaultDriver: DriverPresenter;
    defaultVehicle: VehiclePresenter;
}